### DISCUSSION TITLE
How to create a table form?

### DISCUSSION THREAD

**Discussion by MikeYates** [Budibase/budibase#11642] [Help] [answered]
I am confused how to input table form in budibase design mode and database.

I am creating an accounting app, in my app I am able create a purchase invoice. In this purchase invoice I can enter the invoice date and other information, although I ran into a dead end on how to create a table form to enter item name, quantity, price in one line (also able to add another line when needed) and how to store it into database table.

This is an example how I would want my form to look like :
<img width="937" alt="Screenshot 2023-09-01 at 13 16 22" src="https://github.com/Budibase/budibase/assets/8289655/16dfb10c-cc1e-424a-88ae-b90e72bdb5bd">

**Comment by ConorWebb96** [comment, marked as answer]
Hey @MikeYates,

Is there any reason why you can't just use the Grid component? It likely would have everything but the dynamic form aspect that you are after

If you have a maximum number of line items you may be able to do the above, with [Conditional UI](https://docs.budibase.com/docs/conditions) statements. If you did it this way you'd just have to model your database with the various line items e.g. line item 1, 2, 3, 4 etc. 

If you structure your database well, you can use the `Form Block` component to generate all the fields used within your database table. From here you would then be able to eject and add the additional functionality needed to meet your needs.

You likely won't be able to create dynamic forms like the one in the picture above, there is an open issue here on this: https://github.com/Budibase/budibase/issues/5658

**Reply by MikeYates** [reply]
Hi @ConorWebb96, Thanks for the reply.

Okay Grid component can work although I just realised I need a whole new database for this matter. Because the item list to enter has to be a single database table per purchase invoice. Like your suggestion using line items 1,2,3 etc on a grid would be hard to read and messy, although it seems the current and fastest solution is modelling database table with various line items and using form block.

But doing that we can only set a determine set of lines (eg: 10 line items) and the end user wont be able to manually add another line?
